File: BACK-END/proyectofinal/src/main/java/com/vienteros/proyectofinal/model/Venta.java
package com.vienteros.proyectofinal.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.sql.Date;


@Entity
@Data
@Table(name = "ventas")
public class Venta {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;
    private String nombre;
    private double precio;
    private int cantidad;
    private Date fecha;

    @ManyToOne(targetEntity = Factura.class)
    @JoinColumn(name = "id_factura")
    private Factura factura;

    @ManyToOne(targetEntity = Proyecto.class)
    @JoinColumn(name = "id_proyecto")
    private Proyecto proyecto;
}

File: FRONT-END/proyectofront/src/app/models/venta.ts
import { Factura } from "./factura";
import { Proyecto } from "./proyecto";

export interface Venta{
    id:number;
    nombre: string;
    cantidad: number;
    precio:number;
    fecha: Date;
    data:number;
    proyecto: Proyecto;
    factura:Factura;
}